Mathlib/Analysis/CStarAlgebra/ContinuousFunctionalCalculus/Isometric.lean

```diff
@@ -0,0 +1,488 @@
+/-
+Copyright (c) 2024 Jireh Loreaux. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jireh Loreaux
+-/
+import Mathlib.Analysis.CStarAlgebra.ContinuousFunctionalCalculus.Instances
+
+/-! # Isometric continuous functional calculus
+
+This file adds a class for an *isometric* continuous functional calculus. This is separate from the
+usual `ContinuousFunctionalCalculus` class because we prefer not to require a metric (or a norm) on
+the algebra for reasons discussed in the module documentation for that file.
+
+Of courrse, with a metric on the algebra and an isometric continuous functional calculus, the
+algebra must *be* a C⋆-algebra already. As such, it may seem like this class is not useful. However,
+the main purpose is to allow for the continuous functional calculus to be a isometric for the other
+scalar rings `ℝ` and `ℝ≥0` too.
+-/
+
+local notation "σ" => spectrum
+local notation "σₙ" => quasispectrum
+
+/-! ### Isometric continuous functional calculus for unital algebras -/
+section Unital
+
+/-- An extension of the `ContinuousFunctionalCalculus` requiring that `cfcHom` is an isometry. -/
+class IsometricContinuousFunctionalCalculus (R A : Type*) (p : outParam (A → Prop))
```

Comment:
While we care about different *scalar* rings here, I wonder if we should just require `CStarAlgebra A` everywhere, since these classes will only kick in if `A` is equipped with a metric which necessarily comes from a C⋆-norm.

My thought is that by doing so, we severely limit the search necessary for type class synthessis.